### ISSUE TITLE
Latest version breaks showing metrics on status page

### ISSUE THREAD

**Issue by craftbyte** [cachethq/Cachet#523]
SInce I updated my docker image everytime I add a metric and enable them in settings the status page says "Whoops, looks like something went wrong.".

Error logs are empty at /var/log/nginx/error.log

**Comment by jbrooksuk** [comment]
There was a migration change, but it should've been accounted for in backwards compatibility. What db driver are you using?

**Comment by matejkramny** [comment]
Its using the docker quickstart stepMatej KramnyOn Thu, Mar 19, 2015 at 9:13 am,  James Brooks <notifications@github.com> wrote:            There was a migration change, but it should've been accounted for in backwards compatibility. What db driver are you using?

**Comment by jbrooksuk** [comment]
Oh of course, it'll be Postgres.

**Comment by jbrooksuk** [comment]
These will be the offending lines: https://github.com/cachethq/Cachet/blob/master/src/Models/Metric.php#L83-106

Does the PostgreSQL query look bad? CC @sammcj

**Comment by sammcj** [comment]
Hey @jbrooksuk - Do you have a log of the query that's actually being run?

**Comment by matejkramny** [comment]
@jbrooksuk the docker quickstart uses mysql, so the offending line is 85 and/or 87.

demo.cachethq.io works because heroku uses postgres

**Comment by jbrooksuk** [comment]
I don't think it's that because everything is working locally - I use MySQL to develop Cachet. It could be a broken docker image?

**Comment by bimlendu** [comment]
Probably having the same problem with my install as well. Latest install, the error I am getting is 

```
ErrorException thrown with message "Undefined variable: queryType (View: /usr/share/nginx/html/cachethq/app/views/partials/metrics.blade.php)
```
on this line https://github.com/cachethq/Cachet/blob/master/src/Models/Metric.php#L97

I am using sqlite3 db backend.

**Comment by jbrooksuk** [comment]
Ok, try now.

**Comment by bimlendu** [comment]
Getting a separate error now: 

```
SQLSTATE[HY000]: General error: 1 near "'0 hour'": syntax error (SQL: select sum(metric_points.value) as aggregate FROM metrics JOIN metric_points ON metric_points.metric_id = metrics.id WHERE to_char(metric_points.created_at, 'YYYYMMDDHH') = :timestamp AND to_char(metric_points.created_at, 'H') = to_char(now() - interval '0 hour', 'H') GROUP BY to_char(metric_points.created_at, 'H')) (View: /usr/share/nginx/html/cachethq/app/views/partials/metrics.blade.php) (View: /usr/share/nginx/html/cachethq/app/views/partials/metrics.blade.php)
```

**Comment by bimlendu** [comment]
The query 

```
select sum(metric_points.value) as aggregate FROM metrics JOIN metric_points ON metric_points.metric_id = metrics.id WHERE to_char(metric_points.created_at, 'YYYYMMDDHH') = :timestamp AND to_char(metric_points.created_at, 'H') = to_char(now() - interval '0 hour', 'H') GROUP BY to_char(metric_points.created_at, 'H')
```

uses a few mysql specific functions, like now() or to_char or now() - interval '0 hour', 'H' , which are not there is sqlite and probably these are causing issues.

**Comment by jbrooksuk** [comment]
There is a condition based on your database driver for working that out.

**Comment by jrthib** [comment]
I'm also receiving this error when adding metrics. Any ideas how to fix it?

**Comment by jrthib** [comment]
Seems to be on line 110 of Metric.php. I'm not sure why `$value` is undefined, being outside of the scope that it is defined in, however this seems to be a temporary fix for me until an update or something is pushed.

Replace this:

    if ($value === 0 && $this->default_value != $value) {
        return $this->default_value;
    }

With this:

    if (!isset($value) || ($value === 0 && $this->default_value != $value)) {
        return $this->default_value;
    }

**Comment by jbrooksuk** [comment]
@jrthib do you have a `calc_type` field in your `metrics` tab;e?

**Comment by maltorfer** [comment]
Same issue here using mysql. I do have a calc_type field in my metrics table.

**Comment by jbrooksuk** [comment]
Can you tell me the values that are in that field now for each row?

**Comment by maltorfer** [comment]
only one row and value is 1.

**Comment by jbrooksuk** [comment]
So you want the average value?

**Comment by maltorfer** [comment]
yes

**Comment by maltorfer** [comment]
when you change "average" in the gui and save it, it appears either not being saved or displayed wrong. when you access the metric again, it shows "sum".

**Comment by jrthib** [comment]
Can confirm. I have the calc_type field as well and I'm using mysql. 






On Mon, Mar 23, 2015 at 3:03 AM -0700, "maltorfer" <notifications@github.com> wrote:












when you change "average" in the gui and save it, it appears either not being saved or displayed wrong. when you access the metric again, it shows "sum".

**Comment by urtho** [comment]
Changing this int Metric.php solved my issue with metrics:

=== self::CALC_SUM to == self::CALC_SUM 
and 
=== self::CALC_AVG to == self::CALC_AVG

**Comment by jbrooksuk** [comment]
Closed by #544.
